packages/core/src/components/sections/ProductShelf/ProductShelf.tsx

```diff
@@ -1,89 +1,20 @@
-import { useEffect, useId, useRef } from 'react'
 import { useInView } from 'react-intersection-observer'
-
-import { ProductShelf as UIProductShelf } from '@faststore/ui'
-
-import type { ProductsQueryQueryVariables } from '@generated/graphql'
-import ProductShelfSkeleton from 'src/components/skeletons/ProductShelfSkeleton'
-import { useViewItemListEvent } from 'src/sdk/analytics/hooks/useViewItemListEvent'
-import { useProductsQuery } from 'src/sdk/product/useProductsQuery'
-import { textToKebabCase } from 'src/utils/utilities'
-
-import Carousel from '../../ui/Carousel'
 import Section from '../Section'
-import { Components } from './Overrides'
-const { ProductCard } = Components
 
 import styles from './section.module.scss'
+import UIProductShelf, {
+  ProductShelfProps,
+} from 'src/components/ui/ProductShelf'
 
-interface ProductShelfProps extends Partial<ProductsQueryQueryVariables> {
-  title: string
-  withDivisor?: boolean
-}
-
-function ProductShelf({
-  title,
-  withDivisor = false,
-  ...variables
-}: ProductShelfProps) {
-  const titleId = textToKebabCase(title)
-  const id = useId()
-  const viewedOnce = useRef(false)
+function ProductShelf({ ...otherProps }: Omit<ProductShelfProps, 'inView'>) {
   const { ref, inView } = useInView()
-  const products = useProductsQuery(variables)
-  const productEdges = products?.edges ?? []
-  const aspectRatio = 1
-
-  const { sendViewItemListEvent } = useViewItemListEvent({
-    products: productEdges,
-    title,
-    page: 0,
-    pageSize: 0,
-  })
-
-  useEffect(() => {
-    if (inView && !viewedOnce.current && productEdges.length) {
-      sendViewItemListEvent()
-
-      viewedOnce.current = true
-    }
-  }, [inView, productEdges.length, sendViewItemListEvent])
-
-  if (products?.edges.length === 0) {
-    return null
-  }
 
   return (
     <Section
-      className={`${styles.section} section-product-shelf layout__section ${
-        withDivisor ? 'section__divisor' : ''
```

Comment:
won't we use the `withDivisor` anymore? if so we need to remove the `section__divisor` styles as well .

Reply:
Yes, we don't have this prop mapped in CMS